hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/zlib/BuiltInGzipCompressor.java

```diff
@@ -102,7 +103,15 @@ public int compress(byte[] b, int off, int len) throws IOException {
 
     if (state == BuiltInGzipDecompressor.GzipStateLabel.INFLATE_STREAM) {
       // now compress it into b[]
-      int deflated = deflater.deflate(b, off, len);
+      int deflated;
+      try {
+        deflated = deflater.deflate(b, off, len);
+      } catch (NullPointerException npe) {
```

Comment:
this seems a little fragile since it relies on the error message from another library. Instead, can we add another internal state `ENDED` and check it before? the state can be set in the `end` method of this class.

Reply:
Thank you, I wasn't too comfortable with this either.